server/src/interfaces.ts

```diff
@@ -31,3 +31,36 @@ export interface CustomerPayload {
 export interface CustomerResponse extends CustomerPayload {
   id: number;
 }
+
+/**
+ * ListingPayload Interface that contains the fields of the payload that
+ * would be sent to create a Listing Entity.
+ */
+export interface ListingPayload {
+  merchantId: number;
+  name: string;
+  price: string;
+  oldPrice: string;
```

Comment:
should price be number?

Reply:
@ifhuang Hmm Spot payments doc use string so..... HAHA. Also didn't want to risk floating point errors

Reply:
yeah, no floating number.
I was thinking units and nanos of money.proto: https://github.com/googleapis/googleapis/blob/master/google/type/money.proto

How about we create a money type? With that, the precision supported is clear. Using string, I'm not sure how many digits (after the decimal point) do we support

Reply:
@ifhuang Ok! Made the changes to the datastore to make price an embedded entity:
For example
```
{
  "properties": {
    "currency": {
      "stringValue": "SGD"
    },
    "dollars": {
      "integerValue": 85
    },
    "cents": {
      "integerValue": 0
    }
  }
}
```